src/s1reader/s1_burst_slc.py

```diff
@@ -263,17 +264,31 @@ def __str__(self):
     def __repr__(self):
         return f"{self.__class__.__name__}(burst_id={self.burst_id})"
 
-    def as_isce3_radargrid(self):
+    def as_isce3_radargrid(self, az_step: Optional[float] = None, rg_step: Optional[float] = None):
         '''Init and return isce3.product.RadarGridParameters.
 
-        Returns:
-        --------
+        Parameters
+        ----------
+        az_step : float, optional
+            Azimuth step size in seconds. If not provided, the azimuth step
+            size is set to the azimuth time interval.
+        rg_step : float, optional
+            Range step size in meters. If not provided, the range step size
+            is set to the range pixel spacing.
+
+        Returns
+        -------
         _ : RadarGridParameters
             RadarGridParameters constructed from class members.
         '''
 
+        if az_step is None:
```

Comment:
```suggestion
        length, width = self.shape
        if az_step is None:
            length_in_seconds = length / self.azimuth_time_interval
            if az_step < 0:
                raise ValueError("az_step cannot be negative")
            if 1 / az_step > length_in_seconds:
                raise ValueError("az_step cannot be larger than radar grid")
            length = (length / self.azimuth_time_interval) // az_step
```
Grid length will needs to be recalculated.
Also error checks for sanity. I think I have it right...
Move size extraction up to accommodate error checking.

Reply:
I took the logic, but fixed a couple other things (like overwriting the length/width after this block)